SD-K9/src/app/pages/location-search/location-search.page.ts

```diff
@@ -63,4 +88,31 @@ export class LocationSearchPage implements OnInit {
     }
     this._navController.navigateBack("home");
   }
+
+  moveMap(query: string){
+    this.currentMapState = this._googleStore.getGoogleMapState();
+    let map = this.currentMapState.map;
+    this.coder = this.currentMapState.geocoder;
+
+    this.coder.geocode( { 'address' : query }, function( results, status ) {
+        if( status == google.maps.GeocoderStatus.OK ) {
+          //move map to selected address
+          map.setCenter( results[0].geometry.location );
+          
+          //removes old marker if exist
+          if (typeof this.marker !== 'undefined'){
+            this.marker.setMap(null);
+          }
+          
+          this.marker = new google.maps.Marker( {
+            map     : map,
+            position: results[0].geometry.location
+          } );
+        } else {
+            console.error( 'Geocode was not successful for the following reason: ' + status );
+        }
+    } );
+    this._googleStore.updateGoogleMap(this.currentMapState);
+  }
+
```

Comment:
I think the location search should stick to finding locations and the manipulation of the outdoor map itself should be handled by the outdoor-map component. Also i think there are a few user stories that require the outdoor map to pan to the location indicated in the "start" of the top-directions-bar component. I suggest moving this to the outdoor-map component. By panning the map according to the indicated start location, it will solve those user stories or tasks that depend on that.

Reply:
Good idea! However, I think it can be done in future implementation because of the current workload and leave it as it is.